debezium-core/src/main/java/io/debezium/transforms/ExtractNewRecordState.java

```diff
@@ -105,80 +83,56 @@ public class ExtractNewRecordState<R extends ConnectRecord<R>> implements Transf
 
     @Override
     public void configure(final Map<String, ?> configs) {
-        final Configuration config = Configuration.from(configs);
-        smtManager = new SmtManager<>(config);
-
-        final Field.Set configFields = Field.setOf(ExtractNewRecordStateConfigDefinition.DROP_TOMBSTONES,
-                ExtractNewRecordStateConfigDefinition.HANDLE_DELETES);
-        if (!config.validateAndRecord(configFields, LOGGER::error)) {
-            throw new ConnectException("Unable to validate config.");
-        }
-
-        dropTombstones = config.getBoolean(ExtractNewRecordStateConfigDefinition.DROP_TOMBSTONES);
-        handleDeletes = DeleteHandling.parse(config.getString(ExtractNewRecordStateConfigDefinition.HANDLE_DELETES));
-
-        String addFieldsPrefix = config.getString(ExtractNewRecordStateConfigDefinition.ADD_FIELDS_PREFIX);
-        String addHeadersPrefix = config.getString(ExtractNewRecordStateConfigDefinition.ADD_HEADERS_PREFIX);
-        additionalFields = FieldReference.fromConfiguration(addFieldsPrefix, config.getString(ExtractNewRecordStateConfigDefinition.ADD_FIELDS));
-        additionalHeaders = FieldReference.fromConfiguration(addHeadersPrefix, config.getString(ExtractNewRecordStateConfigDefinition.ADD_HEADERS));
-
-        String routeFieldConfig = config.getString(ExtractNewRecordStateConfigDefinition.ROUTE_BY_FIELD);
-        routeByField = routeFieldConfig.isEmpty() ? null : routeFieldConfig;
+        super.configure(configs);
 
         dropFieldsHeaderName = config.getString(DROP_FIELDS_HEADER);
         dropFieldsFromKey = config.getBoolean(DROP_FIELDS_FROM_KEY);
         dropFieldsKeepSchemaCompatible = config.getBoolean(DROP_FIELDS_KEEP_SCHEMA_COMPATIBLE);
-
-        Map<String, String> delegateConfig = new LinkedHashMap<>();
-        delegateConfig.put("field", "before");
-        beforeDelegate.configure(delegateConfig);
-
-        delegateConfig = new HashMap<>();
-        delegateConfig.put("field", "after");
-        afterDelegate.configure(delegateConfig);
-
-        delegateConfig = new HashMap<>();
-        delegateConfig.put("static.field", ExtractNewRecordStateConfigDefinition.DELETED_FIELD);
-        delegateConfig.put("static.value", "true");
-        removedDelegate.configure(delegateConfig);
-
-        delegateConfig = new HashMap<>();
-        delegateConfig.put("static.field", ExtractNewRecordStateConfigDefinition.DELETED_FIELD);
-        delegateConfig.put("static.value", "false");
-        updatedDelegate.configure(delegateConfig);
-
         schemaUpdateCache = new BoundedConcurrentHashMap<>(SCHEMA_CACHE_SIZE);
     }
 
     @Override
-    public R apply(final R record) {
+    public R doApply(final R record) {
+        // Handling tombstone record
         if (record.value() == null) {
-            if (dropTombstones) {
-                LOGGER.trace("Tombstone {} arrived and requested to be dropped", record.key());
-                return null;
+            if (enableDeleteHandlingV2) {
```

Comment:
Can those `if (enableDeleteHandlingV2)` be converted into methods and placed into strategy interface that would be implemented by configuration enums or similar approach like having a strategy class created based on the config field used? Or maybe even better have the switches removed too if possible?